src/requiredBy.js

```diff
@@ -0,0 +1,12 @@
+export default function requiredBy(requiredByPropName, propType, defaultValue = null) {
+  return function validator(props, propName, componentName, ...rest) {
+    if (props[requiredByPropName]) {
```

Comment:
I wonder if it would be more generic to have `requiredBy(props => ‼props.foo, PropTypes.whatever)` - iow, to have the consumer specify the logic by which it's required, with a predicate? I can't really decide what's better.

However, with the API as-is, this check seems correct. Thanks for explaining!